### ISSUE TITLE
Verify failed transaction

### ISSUE THREAD

**Issue by gantunesr** [MetaMask/metamask-mobile#3165]
### **Description**

Describe the task. What does this aim to solve?

### **Technical Details**

-   Implementation details
-   Insight to what needs to be done
-   Etc.

### **Acceptance Criteria**

-   Cases to satisfy
-   XYZ should work
-   Etc.

### **References**

-   [PR #598](https://github.com/MetaMask/controllers/pull/598)
-   Relevant comment:

> Even a connectivity issue shouldn't make getTransactionByHash return null. I'd expect it to throw in that circumstance. It should only return null if Infura successfully received the response and replied that they don't recognize the hash.
If you're sure that getTransactionByHash returned null for a transaction that later showed up as confirmed, it might have been a fluke of timing. Infura might have legitimately dropped the transaction, only for another node to mine a block that included it. If that's the case, then the bug you saw remains, just it's less likely to encounter it because we're making two calls rather than one. I would bet that calling getTransactionByHash a second time instead of getTransactionReceipt would have identical results.
We had reports of this on the extension as well, and we addressed it by leaving the transaction as "pending" until we saw it was dropped for 3 consecutive blocks. Only after 3 blocks of Infura reporting it as dropped would we tell the user it was dropped. This seemed to reduce the chances of this problem occurring, though I think it's still possible (albeit very rare).
A more robust solution for this probably deserves its own PR, so I certainly don't want to block on that here. But we ought to at least include some comments about why this is done in this manner, because it's really not obvious from reading the code.

**Comment by mobularay** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/MetaMask-Mobile-5f984938ddc0e4001d4b79cb/issues/MetaMask/metamask-mobile/3165?planning-poker) with ZenHub @Cal-L @gantunesr @andrepimenta

**Comment by mobularay** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/MetaMask-Mobile-5f984938ddc0e4001d4b79cb/issues/MetaMask/metamask-mobile/3165?planning-poker) with ZenHub @cortisiko

**Comment by gantunesr** [comment]
Controllers PR: https://github.com/MetaMask/controllers/pull/608

**Comment by mobularay** [comment]
From sprint grooming session on Nov 18: after further assessment, what was implemented could be actually more efficient than having this ticket implemented. Also the work for this ticket requires Infura's help. 

Will be put on hold until Jan 2022 to re-assess again and conclude decision.
